dashboard/models.py

```diff
@@ -38,7 +38,7 @@ def __str__(self):
     # # Replace the year in the end date with start date (Hack to get around far out years)
     # # This should be replaced in the future via an API call so the terms have correct end years, or Canvas data adjusted
     def get_correct_date_end(self):
-        if (self.date_end.year - self.date_start.year) > 1:
+        if self.date_start is not None and (self.date_end.year - self.date_start.year) > 1:
```

Comment:
I'm wondering if we should future-proof this a bit more and also check whether `self.date_end is not None`. `Course.get_course_date_range` checks for this, and we still want to do that there, but this method could also be called in other contexts. Since `date_end` is nullable, that would also break the method.

Adding type hints would also be useful.

Reply:
code updated. Please review.